File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        row, col = len(matrix), len(matrix[0])
        row_ind, col_ind = 0, col - 1
        while(0 <= row_ind < row and 0 <= col_ind < col):
            cur = matrix[row_ind][col_ind]
            if target == cur:
                return True
            elif target < cur:
                col_ind = col_ind - 1
            else:
                row_ind = row_ind + 1
        return False
                
        

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int nrow = matrix.length, ncol = matrix[0].length;
        int row = 0, col = ncol - 1;
        int cur = 0;
        while(row >= 0 && row < nrow && col >= 0 && col < ncol){
            cur = matrix[row][col];
            if(cur == target) return true;
            if(cur > target)col-=1;
            if(cur < target)row+=1;
        }
        return false;
    }
}